micropython/net/ntptime/ntptime.py

```diff
@@ -41,10 +43,9 @@ def time():
     return val - NTP_DELTA
 
 
-# There's currently no timezone support in MicroPython, and the RTC is set in UTC time.
 def settime():
     t = time()
     import machine
 
     tm = utime.gmtime(t)
-    machine.RTC().datetime((tm[0], tm[1], tm[2], tm[6] + 1, tm[3], tm[4], tm[5], 0))
+    machine.RTC().datetime((tm[0], tm[1], tm[2], tm[6] + 1, tm[3] + timezone, tm[4], tm[5], 0))
```

Comment:
Unfortunately, timezone support is far more complicated to add than this appears.

For instance, with this change, if `timezone` is set to +11 and the current UTC hour (`tm[3]`) is 16 (8 pm), this will now try to set the rtc to an hour setting of 25 - this isn't going to work :-(

The rollover < 0 and > 23 would need to be carefully managed. Which then flows through to possible rollovers of day / date as well.